yt_dlp/extractor/bbc.py

```diff
@@ -1069,83 +1099,139 @@ def _real_extract(self, url):
-                    continue
-                identifiers = lead_media.get('identifiers')
-                if not identifiers or not isinstance(identifiers, dict):
-                    continue
-                programme_id = identifiers.get('vpid') or identifiers.get('playablePid')
+            for lead_media in traverse_obj(morph_payload, (
+                    'body', 'components', ..., 'props', 'leadMedia', {dict})):
+                programme_id = traverse_obj(lead_media, ('identifiers', ('vpid', 'playablePid'), {str}, any))
                 if not programme_id:
                     continue
-                title = lead_media.get('title') or self._og_search_title(webpage)
                 formats, subtitles = self._download_media_selector(programme_id)
-                description = lead_media.get('summary')
-                uploader = lead_media.get('masterBrand')
-                uploader_id = lead_media.get('mid')
-                duration = None
-                duration_d = lead_media.get('duration')
-                if isinstance(duration_d, dict):
-                    duration = parse_duration(dict_get(
-                        duration_d, ('rawDuration', 'formattedDuration', 'spokenDuration')))
                 return {
                     'id': programme_id,
-                    'title': title,
-                    'description': description,
-                    'duration': duration,
-                    'uploader': uploader,
-                    'uploader_id': uploader_id,
+                    'title': lead_media.get('title') or self._og_search_title(webpage),
+                    **traverse_obj(lead_media, {
+                        'description': ('summary', {str}),
+                        'duration': ('duration', ('rawDuration', 'formattedDuration', 'spokenDuration'), {parse_duration}),
+                        'uploader': ('masterBrand', {str}),
+                        'uploader_id': ('mid', {str}),
+                    }),
                     'formats': formats,
                     'subtitles': subtitles,
                 }
+            body = self._parse_json(traverse_obj(morph_payload, (
+                'body', 'content', 'article', 'body')), playlist_id, fatal=False)
+            for video_data in traverse_obj(body, (lambda _, v: v['videoData']['pid'], 'videoData')):
+                if video_data.get('vpid'):
+                    video_id = video_data['vpid']
+                    formats, subtitles = self._download_media_selector(video_id)
+                    entry = {
+                        'id': video_id,
+                        'formats': formats,
+                        'subtitles': subtitles,
+                    }
+                else:
+                    video_id = video_data['pid']
+                    entry = self.url_result(
+                        f'https://www.bbc.co.uk/programmes/{video_id}', BBCCoUkIE,
+                        video_id, url_transparent=True)
+                entry = {
```

Comment:
You missed
```suggestion
                entry = {
                    **entry,
```
or probably better
```suggestion
                entry.update({
```
Does none of the tests cover this code path?

Reply:
This was supposed to provide common values for `entry` from metadata found elsewhere:
```py
                entry = merge_dicts(
                    traverse_obj(morph_payload, (
                        'body', 'content', 'article', {
                            'timestamp': ('dateTimeInfo', 'dateTime', {parse_iso8601}),
                        })),
                    traverse_obj(video_data, {
                        'thumbnail': (('iChefImage', 'image'), {url_or_none}),
                        'title': (('title', 'caption'), {str}), any),
                        'duration': ('duration', {parse_duration}),
                    }),
                    entry)
```
As there is no defaulting, the constructs `entry = {**t_o(...),**t_o(...), **entry}` or `entry.update({**t_o(...),**t_o(...)}` are equivalent.

The noplaylist test using http://www.bbc.com/sport/0/football/34475836 should apply.

Reply:
ok, manually running the noplaylist test. Not sure why this didn't catch before.